File: ThunderBasics/CAGradientLayer+AutoGradient.m
//
//  CAGradientLayer+AutoGradient.m
//  ThunderStorm
//
//  Created by Andrew Hart on 20/11/2013.
//  Copyright (c) 2013 3 SIDED CUBE. All rights reserved.
//

#import "CAGradientLayer+AutoGradient.h"

@implementation CAGradientLayer (AutoGradient)

+ (CAGradientLayer *)generateGradientLayerWithTopColor:(UIColor *)topColor bottomColor:(UIColor *)bottomColor
{
    NSArray *colors = [NSArray arrayWithObjects:(id)topColor.CGColor, bottomColor.CGColor, nil];
    NSNumber *stopOne = [NSNumber numberWithFloat:0.0];
    NSNumber *stopTwo = [NSNumber numberWithFloat:1.0];
    
    NSArray *locations = [NSArray arrayWithObjects:stopOne, stopTwo, nil];
    
    CAGradientLayer *headerLayer = [CAGradientLayer layer];
    headerLayer.colors = colors;
    headerLayer.locations = locations;
    
    return headerLayer;
}

@end

File: ThunderBasics/CAGradientLayer+AutoGradient.swift
//
//  CAGradientLayer+AutoGradient.swift
//  ThunderBasics
//
//  Created by Simon Mitchell on 05/11/2018.
//  Copyright © 2018 threesidedcube. All rights reserved.
//

import UIKit

public extension CAGradientLayer {
    
    /// Creates a `CAGradientLayer` transitioning from one color to another.
    /// A transparent color can be parsed to create dark to translucent gradients.
    ///
    /// - Parameters:
    ///   - topColor: The gradients top color represented as a `UIColor`. Starts at location 0.0
    ///   - bottomColor: The gradients bottom color represented as a `UIColor`. Starts at location 1.0
    convenience init(topColor: UIColor, bottomColor: UIColor) {
        self.init()
        colors = [topColor.cgColor, bottomColor.cgColor]
        locations = [0.0, 1.0].map({ NSNumber(value: $0) })
    }
}
